ground/gcs/src/plugins/uavsettingsimportexport/importsummary.cpp

```diff
@@ -37,14 +54,34 @@ ImportSummaryDialog::ImportSummaryDialog( QWidget *parent) :
    ui->importSummaryList->setColumnCount(3);
    ui->importSummaryList->setRowCount(0);
    QStringList header;
-   header.append("Save");
+   header.append("Use");
    header.append("Name");
    header.append("Status");
    ui->importSummaryList->setHorizontalHeaderLabels(header);
    ui->progressBar->setValue(0);
 
    connect( ui->closeButton, SIGNAL(clicked()), this, SLOT(close()));
-   connect(ui->saveToFlash, SIGNAL(clicked()), this, SLOT(doTheSaving()));
+
+   // Connect the Select All/None buttons
+   QSignalMapper* signalMapper = new QSignalMapper(this);
+
+   connect(ui->btnSaveToFlash, SIGNAL(clicked()), signalMapper, SLOT(map()));
+   connect(ui->btnApply, SIGNAL(clicked()), signalMapper, SLOT(map()));
+
+   signalMapper->setMapping(ui->btnSaveToFlash, UAVSettingsAction::save);
+   signalMapper->setMapping(ui->btnApply, UAVSettingsAction::apply);
+
+   connect(signalMapper, SIGNAL(mapped(int)), this, SLOT(doTheApplySaving(int)));
+
+   // Connect the Select All/None buttons
+   signalMapper = new QSignalMapper (this);
```

Comment:
This also looks like lost memory. I don't see why not to declare these in the class.